src/lsm/level_data_iterator.zig

```diff
@@ -90,6 +90,19 @@ pub fn LevelDataIteratorType(comptime Table: type, comptime Storage: type) type
             it.* = undefined;
         }
 
+        pub fn reset(it: *LevelDataIterator) void {
+            std.mem.set(u8, it.index_block, 0);
+            it.level_index_iterator.reset();
+            it.table_data_iterator.reset();
```

Comment:
`deinit` uses reverse-init order. It isn't strictly required in this case, but I think it is a good convention to use, since there are cases where it _is_ required.